### PR TITLE
Soc 2677

### PR DESCRIPTION
**Pull request by qaga**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+package com.wikia.webdriver.elements.mercury.components.loginAndSignup;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java]: MAJOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00120)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  @FindBy(css = "#signupForm > small.error")
+  private WebElement genericError;
+  @FindBy(css = " header.auth-header")
+  private WebElement registerHeader;
+  @FindBy(css = ".loginAndSignup")
+  private WebElement newloginButton;
+  @FindBy(css = ".signup-provider-email")
+  private WebElement signupButton;
+
+//  private WebDriver driver;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+//  private Wait wait;
+  private UrlBuilder urlBuilder;
+
+  private final String mainWindowHandle;
+
+
+  public RegisterArea(WebDriver driver) {
+//    this.driver = driver;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/LoginArea.java [hunk 4]
```diff
@@ -1,14 +1,16 @@
-package com.wikia.webdriver.elements.mercury.components.login;
+package com.wikia.webdriver.elements.mercury.components.loginAndSignup;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/LoginArea.java]: MAJOR Rename this package name to match the regular expression '^[a-z]+(\.[a-z][a-z0-9]*)*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00120)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [3 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/1b0c8fe32fdf4f33a325863205fb9970f7bf846b/src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/login/SignInPage.java [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.core.elemnt.Wait;
-import com.wikia.webdriver.elements.mercury.components.login.LoginArea;
+import com.wikia.webdriver.elements.mercury.components.loginAndSignup.LoginArea;
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/login/SignInPage.java]: Please split these two to login and ignup area:  
LoginArea goes to components.login.
RegisterArea goest to components.signup

File: src/test/java/com/wikia/webdriver/elements/oasis/components/notifications/BannerNotifications.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  @FindBy(css = ".banner-notification div.msg a")
+  private WebElement undeleteLink;
+  @FindBy(css = ".banner-notification")
+  private WebElement bannerNotification;
+  @FindBy(css = ".banner-notification.non-dismissible")
+  private WebElement bannerNotificationEmailNotConfirmed;
+
+
+  public SpecialRestorePageObject undeleteByFlashMessage() {
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/notifications/BannerNotifications.java]: what does it mean undeleteByFlashMessage ? Can it be more descriptive ?

File: src/test/java/com/wikia/webdriver/elements/oasis/components/notifications/BannerNotifications.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    wait.forElementVisible(undeleteLink);
+    undeleteLink.click();
+    return new SpecialRestorePageObject(driver);
+  }
+
+  public void verifyNotificationMessage() {
+    driver.manage().timeouts().implicitlyWait(250, TimeUnit.MILLISECONDS);
+    try {
+      wait.forElementVisible(bannerNotification);
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/elements/oasis/components/notifications/BannerNotifications.java]: why does it need the implicit wait ? Can't just wait.forElementVisible be used ?

If not, such method might be there already

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [3 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/b2dcc627d9b11ba19285d9d9e95bbd3d6f9dbd5b/src/test/java/com/wikia/webdriver/elements/mercury/components/loginAndSignup/RegisterArea.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [3 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a8aa941d9d270710b9020eff60b336571b5ea8df/src/test/java/com/wikia/webdriver/elements/mercury/components/signup/RegisterArea.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [3 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/07b90dd5996cb5f46ed2e6d59114bfff8eac96c0/src/test/java/com/wikia/webdriver/elements/mercury/components/signup/RegisterArea.java) [rule](http://sonar-s1/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java [hunk 8]
```diff
@@ -1,154 +1,155 @@
...
+    register.typeEmailAddress(email);
+    register.typeUsername(userName);
+    register.typePassword(password);
+    register.typeBirthdate(PageContent.WIKI_SIGN_UP_BIRTHMONTH, PageContent.WIKI_SIGN_UP_BIRTHDAY,
+                           PageContent.WIKI_SIGN_UP_BIRTHYEAR);
+    register.clickSignUpSubmitButton();
+    base.verifyUserLoggedIn(userName);
+    BannerNotifications notification = new BannerNotifications();
+    Assertion.assertEquals(notification.getBannerNotificationTextEmailNotConfirmed(),
+                           "メールアドレスの認証が完了していないようです。受信トレイの確認メールをチェックしてみてください。確認メールが見つからない場合は、ここをクリックすると新しい確認メールが送信されます。メールアドレスを変更する必要がある場合は、「個人設定」ページにアクセスしてください。");
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java]: make the japanese string a variable and make it shorter. It's too long

File: src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java [hunk 9]
```diff
@@ -1,154 +1,155 @@
...
+    register.typeEmailAddress(email);
+    register.typeUsername(userName);
+    register.typePassword(password);
+    register.typeBirthdate(PageContent.WIKI_SIGN_UP_BIRTHMONTH, PageContent.WIKI_SIGN_UP_BIRTHDAY,
+                           PageContent.WIKI_SIGN_UP_BIRTHYEAR);
+    register.clickSignUpSubmitButton();
+    base.verifyUserLoggedIn(userName);
+    BannerNotifications notification = new BannerNotifications();
+    Assertion.assertEquals(notification.getBannerNotificationTextEmailNotConfirmed(),
+                           "Oh no! Your email address has not yet been confirmed. You should have a confirmation message in your inbox. Didn't get it? Click here and we'll send a new one. If you need to change your address, head to your Preferences page.");
```
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/signuptests/SignUpTests.java]: Make this string variable and shorter

**Comment by justnpT** [comment]
lgtm

**Pull request merged**